lib/server.js

```diff
@@ -254,6 +255,19 @@ module.exports = class Server extends EventEmitter {
       return hs
     }
 
+    if (this._shareLocalAddress && !direct && clientAddress.host === serverAddress.host) {
+      const clientAddresses = remotePayload.addresses4.filter(onlyPrivateHosts)
+
+      if (clientAddresses.length > 0) {
```

Comment:
Replying the previous review, it doesn't seem necessary to guard against serverSocket not existing